wallet/common/symbol/src/utils/account.js

```diff
@@ -0,0 +1,163 @@
+	const address = facade.network.publicKeyToAddress(_publicKey);
+
+	const account = {
+		address: address.toString(),
+		publicKey,
+		networkIdentifier
+	};
+
+	if (index !== null && index !== undefined)
+		account.index = index;
+
+	if (accountType)
+		account.accountType = accountType;
+
+	return account;
+};
+
+/**
+ * Creates the public account from a given private key.
+ * @param {string} privateKey - The private key.
+ * @param {string} networkIdentifier - The network identifier.
+ * @param {string} [accountType] - The account type (optional).
+ * @param {number} [index] - The account index (optional).
+ * @returns {PublicAccount} The public account.
+ */
+export const publicAccountFromPrivateKey = (privateKey, networkIdentifier, accountType, index) => {
+	const _privateKey = new PrivateKey(privateKey);
+	const keyPair = new SymbolFacade.KeyPair(_privateKey);
+	const publicKey = keyPair.publicKey.toString();
+
+	return publicAccountFromPublicKey(publicKey, networkIdentifier, accountType, index);
+};
+
+/**
+ * Creates a wallet storage account object. This object includes the private key.
+ * @param {string} privateKey - The private key.
+ * @param {string} networkIdentifier - The network identifier.
+ * @param {string} [accountType] - The account type (optional).
+ * @param {number} [index] - The account index (optional).
+ * @returns {PrivateAccount} The wallet storage account object.
+ */
+export const createPrivateAccount = (privateKey, networkIdentifier, accountType, index) => {
+	const publicAccount = publicAccountFromPrivateKey(privateKey, networkIdentifier, accountType, index);
+
+	return {
+		...publicAccount,
+		privateKey
+	};
+};
+
+/**
+ * Creates a wallet account object.
+ * @param {string} privateKey - The private key.
+ * @param {string} networkIdentifier - The network identifier.
+ * @param {string} name - The account name.
+ * @param {string} accountType - The account type.
+ * @param {number} [index] - The account index.
+ * @returns {WalletAccount} The wallet account object.
+ */
+export const createWalletAccount = (privateKey, networkIdentifier, name, accountType, index) => {
+	const publicAccount = publicAccountFromPrivateKey(privateKey, networkIdentifier, accountType, index);
+
+	return {
+		...publicAccount,
+		name,
+		accountType
+	};
+};
+
+/**
+ * Checks if a given string is a valid private key.
+ * @param {string} stringToTest - The string to test.
+ * @returns {boolean} A boolean indicating if the string is a valid private key.
+ */
+export const isPrivateKey = stringToTest => {
+	if (typeof stringToTest !== 'string')
+		return false;
+
+	try {
+		new PrivateKey(stringToTest);
+		
+		return true;
+	} catch {
+		return false;
+	}
+};
+
+/**
+ * Checks if a given string is a valid Symbol address.
+ * @param {string} address - The address to test.
+ * @returns {boolean} A boolean indicating if the string is a valid Symbol address.
+ */
+export const isSymbolAddress = address => {
```

Comment:
Not sure if possible to pass network identifier so it only check address on a specific network.